### PR TITLE
repository-browser error handling

### PR DESCRIPTION
**Pull request by Ke-Jin-Laserfiche**

### CODE REVIEW

File: projects/ui-components/lf-repository-browser/lf-repository-browser.component.ts [hunk 1]
```diff
@@ -70,6 +74,7 @@ export class LfRepositoryBrowserComponent implements OnDestroy {
   @Input()
   refreshAsync: () => Promise<void> = async () => {
     if (!this._currentFolder) {
+      this.isLoading = true;
```
Comment (alexgomezlf) [inline projects/ui-components/lf-repository-browser/lf-repository-browser.component.ts:80]: there is no place where isLoading is set to false?
or is that in a later call?

do we need to initialize breadcrumbs only if there was no currentFolder

This is a more general question, but if there is no currentFolder do we want to try to get the root, or just throw?

File: projects/ui-components/lf-repository-browser/lf-repository-browser.component.ts [hunk 2]
```diff
@@ -351,12 +357,23 @@ export class LfRepositoryBrowserComponent implements OnDestroy {
...
+        this.isLoading = false;
+        this.ref.detectChanges();
         return;
       }
     }
     // If the entry passed in is not a container we will get the parent of this by default.
     if (currentEntry && !currentEntry.isContainer) {
-      currentEntry = await this.treeNodeService.getParentTreeNodeAsync(currentEntry);
+      try {
+        currentEntry = await this.treeNodeService.getParentTreeNodeAsync(currentEntry);
```
Comment (alexgomezlf) [inline projects/ui-components/lf-repository-browser/lf-repository-browser.component.ts]: this.hasError = false should be first in the try

File: projects/ui-components/lf-repository-browser/lf-repository-browser.component.ts [hunk 3]
```diff
@@ -351,12 +357,23 @@ export class LfRepositoryBrowserComponent implements OnDestroy {
       } catch (err: any) {
         console.error(`Error retrieving root node`, JSON.stringify(err));
         this.hasError = true;
```
Comment (alexgomezlf) [inline projects/ui-components/lf-repository-browser/lf-repository-browser.component.ts]: online 350, does this.isLoading need to be set to false?

File: projects/ui-components/lf-repository-browser/lf-repository-browser.component.ts [hunk 4]
```diff
@@ -377,12 +394,16 @@ export class LfRepositoryBrowserComponent implements OnDestroy {
     while (currentNode) {
       try {
         const nextParent: LfTreeNode | undefined = await this.treeNodeService.getParentTreeNodeAsync(currentNode);
+        this.hasError = false;
```
Comment (alexgomezlf) [inline projects/ui-components/lf-repository-browser/lf-repository-browser.component.ts]: this.hasError should be first

File: projects/ui-components/lf-repository-browser/lf-repository-browser.component.ts (hunk 1 above)
Comment (Ke-Jin-Laserfiche) [reply projects/ui-components/lf-repository-browser/lf-repository-browser.component.ts:80]: yeah if there is no currentFolder, we try to get the root, but if there is still no currentFolder, we throw.
should we display error instead? (and add a try catch block for the getRootTreeNodeAsync call)

File: projects/ui-components/lf-repository-browser/lf-repository-browser.component.ts [hunk 5]
```diff
@@ -459,16 +465,17 @@ export class LfRepositoryBrowserComponent implements OnDestroy {
     if (parentEntry && parentEntry.id) {
       try {
         this.isLoading = true;
+        this.hasError = false;
```
Comment (Ke-Jin-Laserfiche) [inline projects/ui-components/lf-repository-browser/lf-repository-browser.component.ts:473]: this function is called as part of initAsync, but also openChildFolderAsync, so I add a separate try/catch for it

File: projects/ui-components/lf-repository-browser/lf-repository-browser.component.ts [hunk 6]
```diff
@@ -70,7 +76,18 @@ export class LfRepositoryBrowserComponent implements OnDestroy {
   @Input()
   refreshAsync: () => Promise<void> = async () => {
     if (!this._currentFolder) {
-      this._currentFolder = await this.treeNodeService.getRootTreeNodeAsync();
+      try {
+        this.hasError = false;
```
Comment (alexgomezlf) [inline projects/ui-components/lf-repository-browser/lf-repository-browser.component.ts]: it looks like this is now all in the if (!this._currentFolder) {}

File: projects/ui-components/lf-repository-browser/lf-repository-browser.component.ts [hunk 7]
```diff
@@ -69,17 +75,26 @@ export class LfRepositoryBrowserComponent implements OnDestroy {
 
   @Input()
   refreshAsync: () => Promise<void> = async () => {
-    if (!this._currentFolder) {
+    try {
+      this.hasError = false;
```
Comment (alexgomezlf) [inline projects/ui-components/lf-repository-browser/lf-repository-browser.component.ts:79]: do we need to set isLoading to true?

**Review by alexgomezlf — DISMISSED** [review]

File: projects/ui-components/lf-repository-browser/lf-repository-browser.component.ts [hunk 8]
```diff
@@ -479,10 +486,19 @@ export class LfRepositoryBrowserComponent implements OnDestroy {
...
+      this.lastDataCall = this.updateFolderChildrenAsync(parentEntry);
+      const selectable = await this.lastDataCall;
+      this.lastDataCall = undefined;
+      return selectable;
+    }
+    catch (error) {
+      this.hasError = true;
+      this.isLoading = false;
+      this.ref.detectChanges();
+      throw error;
```
Comment (alexgomezlf) [inline projects/ui-components/lf-repository-browser/lf-repository-browser.component.ts]: why do you still need to throw the error?

Comment (Ke-Jin-Laserfiche) [reply projects/ui-components/lf-repository-browser/lf-repository-browser.component.ts]: this was to not change the function signature to return undefined. Is there a another way around this?

**Review by alexgomezlf — DISMISSED** [review]

**Review by alexgomezlf — DISMISSED** [review]

**Review by alexgomezlf — APPROVED** [review]

**Pull request merged**